File: backend/internal/constants/deviation_guideword.go
package constants

// DeviationGuideword identifies the standard HAZOP deviation prompt.
type DeviationGuideword string

const (
	GuidewordNo      DeviationGuideword = "no"
	GuidewordMore    DeviationGuideword = "more"
	GuidewordLess    DeviationGuideword = "less"
	GuidewordReverse DeviationGuideword = "reverse"
	GuidewordOther   DeviationGuideword = "other"
)

var deviationGuidewords = map[DeviationGuideword]struct{}{
	GuidewordNo: {}, GuidewordMore: {}, GuidewordLess: {},
	GuidewordReverse: {}, GuidewordOther: {},
}

func (g DeviationGuideword) Valid() bool {
	_, ok := deviationGuidewords[g]
	return ok
}

func DeviationGuidewordValues() []string {
	return []string{string(GuidewordNo), string(GuidewordMore), string(GuidewordLess), string(GuidewordReverse), string(GuidewordOther)}
}

File: frontend/src/types/enums/deviation-guideword.ts
export const deviationGuidewords = ['no', 'more', 'less', 'reverse', 'other'] as const
export type DeviationGuideword = (typeof deviationGuidewords)[number]

export const deviationGuidewordLabels: Record<DeviationGuideword, string> = {
  no: '无 / NO',
  more: '过量 / MORE',
  less: '不足 / LESS',
  reverse: '逆向 / REVERSE',
  other: '其他 / OTHER',
}
